crates/lox_py/src/bodies.rs

```diff
@@ -0,0 +1,314 @@
+/*
+ * Copyright (c) 2024. Helge Eichhorn and the LOX contributors
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, you can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+use pyo3::prelude::*;
+
+use lox_core::bodies::*;
+
+use crate::LoxPyError;
+
+#[pyclass(name = "Sun")]
+pub struct PySun;
+
+#[pymethods]
+impl PySun {
+    #[new]
+    fn new() -> Self {
+        Self
+    }
+
+    fn __repr__(&self) -> PyResult<String> {
+        Ok("Sun()".to_string())
+    }
+
+    fn __str__(&self) -> PyResult<String> {
+        Ok("Sun".to_string())
+    }
+
+    fn id(&self) -> i32 {
+        Sun.id().0
+    }
+
+    fn name(&self) -> &'static str {
+        Sun.name()
+    }
+
+    fn gravitational_parameter(&self) -> f64 {
+        Sun.gravitational_parameter()
+    }
+
+    fn mean_radius(&self) -> f64 {
+        Sun.mean_radius()
+    }
+
+    fn polar_radius(&self) -> f64 {
+        Sun.polar_radius()
+    }
+
+    fn equatorial_radius(&self) -> f64 {
+        Sun.equatorial_radius()
+    }
+}
+
+#[pyclass(name = "Barycenter")]
+pub struct PyBarycenter(Box<dyn PointMass + Send>);
+
+#[pymethods]
+impl PyBarycenter {
+    #[new]
+    fn new(name: &str) -> Result<Self, LoxPyError> {
+        let barycenter: Option<Box<dyn PointMass + Send>> = match name {
+            "ssb" | "SSB" | "solar system barycenter" | "Solar System Barycenter" => {
+                Some(Box::new(SolarSystemBarycenter))
+            }
+            "mercury barycenter" | "Mercury Barycenter" => Some(Box::new(MercuryBarycenter)),
+            "venus barycenter" | "Venus Barycenter" => Some(Box::new(VenusBarycenter)),
+            "earth barycenter" | "Earth Barycenter" => Some(Box::new(EarthBarycenter)),
+            "mars barycenter" | "Mars Barycenter" => Some(Box::new(MarsBarycenter)),
+            "jupiter barycenter" | "Jupiter Barycenter" => Some(Box::new(JupiterBarycenter)),
+            "saturn barycenter" | "Saturn Barycenter" => Some(Box::new(SaturnBarycenter)),
+            "uranus barycenter" | "Uranus Barycenter" => Some(Box::new(UranusBarycenter)),
+            "neptune barycenter" | "Neptune Barycenter" => Some(Box::new(NeptuneBarycenter)),
+            "pluto barycenter" | "Pluto Barycenter" => Some(Box::new(PlutoBarycenter)),
+            _ => None,
+        };
+        match barycenter {
+            Some(barycenter) => Ok(Self(barycenter)),
+            None => Err(LoxPyError::InvalidBody(name.to_string())),
+        }
+    }
+
+    fn __repr__(&self) -> PyResult<String> {
+        Ok(format!("MinorBody(\"{}\")", self.name()))
+    }
+
+    fn __str__(&self) -> PyResult<String> {
+        Ok(self.name().to_string())
+    }
+
+    fn id(&self) -> i32 {
+        self.0.id().0
```

Comment:
IIUC, since everything in Python is an object, it cannot handle a plain C integer and the return value needs to wrapped. They provide `TryFrom` implementations for common Rust types.